### DISCUSSION TITLE
Is there a way to set font_family of social card dynamically depending on the language of content?

### DISCUSSION THREAD

**Discussion by JinIgarashi** [squidfunk/mkdocs-material#5704] [Q&A]
I found the following solution to set font_family in social card for specific language

https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#choosing-a-font

There is an issue of font of social card in QField documentation which has multiple languages' contents.

https://github.com/opengisch/QField-docs/issues/355

Default roboto font does not work for Chinese and Japanese. If we select `Noto Sans JP`, it does not work for Chinese. `Noto Sans SC` also does not work for Japanese.

This QField plugin uses `i18n` plugin to implement internationalisation. Is there a way to dynamically change font_family of social plugin depending on the language of contents? Do you have any advice for this issue?

**Comment by JinIgarashi** [comment]
I just found my commiting QField has likely access to your insider version of module. I think OpenGIS.ch is your regular sponsor.
I guess this can be solved by using customization functionality which is available in insiders by adding font family for each `*.ja.md` and `*.zh.md`.

https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#typography

I am asking OpenGIS.ch to try this customization feature in their issue since I don't have access to insiders.

**Reply by squidfunk** [reply]
Jup, this can definitely be solved by customizing social cards! If you have access to Insiders, I think the simplest way is to use the [built-in `meta` plugin](https://squidfunk.github.io/mkdocs-material/reference/#built-in-meta-plugin) to override [`font_family`](https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#+social.cards_layout_options.font_family) for each subtree of your documentation [as described here](https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#parametrizing-the-layout), something like:

`docs/ja/.meta.yml`

``` yaml
social:
  cards_layout_options:
    font_family: Noto Sans JP
```

`docs/zh/.meta.yml`

``` yaml
social:
  cards_layout_options:
    font_family: Noto Sans SC
```

Note that paths depend on your directory layout. It's currently not supported to scope the application of metadata through `.meta.yml` files based on file extension, i.e., apply metadata only to files ending in `*.zh.md`.
